cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/ExecutorStatusTest.java

```diff
@@ -0,0 +1,123 @@
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        if (state != null) {
+            objectNode.put(STATE_KEY, state);
+        }
+        if (finishedDataMovement != null) {
+            objectNode.put(FINISHED_DATA_MOVEMENT_KEY, finishedDataMovement);
+        }
+        if (totalDataToMove != null) {
+            objectNode.put(TOTAL_DATA_TO_MOVE_KEY, totalDataToMove);
+        }
+        if (triggeredTaskReason != null) {
+            objectNode.put(TRIGGERED_TASK_REASON_KEY, triggeredTaskReason);
+        }
+        return objectNode;
+    }
+
+    @Test
+    public void testIsInProgressState() {
+        ExecutorStatus es0 = new ExecutorStatus(createExecutorStatusJson(
+                CruiseControlExecutorState.NO_TASK_IN_PROGRESS.toString(), null, null, null));
+        assertThat(es0.isInProgressState(), is(false));
+
+        ExecutorStatus es1 = new ExecutorStatus(createExecutorStatusJson(
+                CruiseControlExecutorState.INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS.toString(),
+                DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE, DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es1.isInProgressState(), is(true));
+    }
+
+    @Test
+    public void testGetFinishedDataMovement() {
+        ExecutorStatus es0 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, "50", DEFAULT_TOTAL_DATA_TO_MOVE, DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es0.getFinishedDataMovement(), is(50));
+
+        // Test missing field value is zero
+        ExecutorStatus es1 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, "", DEFAULT_TOTAL_DATA_TO_MOVE, DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es1.getFinishedDataMovement(), is(0));
+
+        // Test missing field throws NoSuchFieldException
+        assertThrows(IllegalArgumentException.class, () ->
+                new ExecutorStatus(createExecutorStatusJson(DEFAULT_STATE, null, null, null)));
+    }
+
+    @Test
+    public void testGetTotalDataToMove() {
+        ExecutorStatus es0 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, "10000", DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es0.getTotalDataToMove(), is(10000));
+
+        // Test missing field value is zero
+        ExecutorStatus es1 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, "", DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es1.getTotalDataToMove(), is(0));
+
+        // Test missing field throws IllegalArgumentException
+        assertThrows(IllegalArgumentException.class, () ->
+                new ExecutorStatus(createExecutorStatusJson(null, null, null, null)));
```

Comment:
This does not assert that the `totalDataToMove` field is present. The state is `null` so it's `extractState()` that throws and not `extractTotalDataToMove()`.